docs/docsite/rst/dev_guide/migrating_roles.rst

```diff
@@ -0,0 +1,345 @@
+
+.. code-block:: bash
+
+    namespace/
+    └── name/
+      ├── docs/
+      ├── galaxy.yml
+      ├── plugins/
+      │   ├── modules/
+      │   │   └── module1.py
+      │   ├── inventory/
+      │   └── .../
+      ├── README.md
+      ├── roles/
+      │   ├── role1/
+      │   ├── role2/
+      │   └── .../
+      ├── playbooks/
+      │   ├── files/
+      │   ├── vars/
+      │   ├── templates/
+      │   └── tasks/
+      └── tests/
+
+A collection can contain one or more roles in the :file:`roles/` directory and these are almost identical to standalone roles with the following exceptions:
+
++----------------------------------------+------------------------------------------------------------------------------+
+| standalone role                        | collection role                                                              |
++========================================+==============================================================================+
+| :file:`library/` with custom modules   | - Move to :file:`plugins/modules/`                                           |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+|``ansible.module_utils`` in Python      + Use :abbr:`FQCN (Fully Qualified Collection Name)` for Python imports        |
++----------------------------------------+------------------------------------------------------------------------------+
+| Plugins in :file:`PLUGINTYPE_plugins/` + - Move to :file:`plugins/PLUGINTYPE/`  (such as :file:`plugins/filter/`)     |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+| :file:`module_utils/`                  + - Move to :file:`plugins/module_utils/`                                      |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+
+.. note::
+
+	 In standalone roles, some of the plugin directories referenced their plugin types in the plural sense; this is not the case in collections.
+
+Migrating a role to a collection
+================================
+
+To migrate from a standalone role to a collection role:
+
+
+1. Create a collection. You need a `Galaxy namespace <https://galaxy.ansible.com/docs/contributing/namespaces.html>`_ to import this collection to Galaxy.
+
+.. code-block:: bash
+
+  $ ansible-galaxy collection init mynamespace.mycollection
```

Comment:
I ran into an issue here—since this command doesn't respect the `collections_paths` you have configured, it will dump a collection structure into `./mynamespace/mycollection`. Since that is not in a collections_paths with the magic folder `ansible_collections`, any playbooks you have locally that refer to the new collection won't be able to find it—unless you manually create an `ansible_collections` directory in a `collections_paths` location (which is not documented), then change directories to that location, then continue with this guide from that point onward.

Reply:
we can distinguish 'setting up collection for dev' vs execution of said collection (which needs to be in known execution path with 'magic dirs' on top.

Reply:
not sure what to change in the doc here?